vulnerabilities/models.py

```diff
@@ -111,68 +164,17 @@ def __str__(self):
         return self.package_url
 
 
-class ImpactedPackage(models.Model):
+class Vulnerability_Package_Relation(models.Model):
```

Comment:
I assume this name is a placeholder, right? How about `VulnerabilityImpact`? Not great, not terrible, IMHO.

Reply:
`VulnerabilityImpact` doesn't mention package anywhere, we need a name which should make sense that the table is about vulnerability and package

Reply:
Yeah that would be better.

Reply:
@sbs2001 I am not convinced by the name too (and we should not use snake case for Class or model names).
The attributes are about a Package (is_vulnerable and version_range are all about Package) so a better name could might PackageAssignedVulnerability or PackageRelatedVulnerability ... but I need to think more about that change as what are the benefits to combine the Impacted and Resolved models in one?

Reply:
@pombredanne 

About the name thing, it's just a placeholder, Btw `PackageRelatedVulnerability` makes tad more sense here. 

>what are the benefits to combine the Impacted and Resolved models in one?

Good question. There are some issues with  having 2 tables, Impacted and Resolved . 

Issue 1 : 
You can do this, which doesn't make any sense. 
```
In [1]: from vulnerabilities import models                                                                                   

In [2]: v1 = models.Vulnerability.objects.create(cve_id="CVE-foo")                                                           

In [3]: p1 = models.Package.objects.create(name="cream",type="ice",version='mango') 
   
In [4]: vp1 = models.ImpactedPackage.objects.create(vulnerability=v1, package=p1)                                            

In [5]: vp2 = models.ResolvedPackage.objects.create(vulnerability=v1, package=p1)    
``` 
This is pure garbage, nothing can be interpreted from these entries. 

With a single table + flag, I can use a `unique_together=('vulnerability','package')`

Issue 2 :  Check  https://github.com/nexB/vulnerablecode/blob/58d0376e7319d06387662cb393f3c39d9893088d/vulnerabilities/import_runner.py#L121 , I am not sure I understand the exact issue but it's something along the lines  that  updating vulnerability status of a already existing package is not possible. @haikoschol  can you explain this, with a snippet? 

Having a single table, changes delete to an update(of the flag), which bypasses this issue.

Reply:
Never mind @haikoschol  already left a brief explanation of this issue in the comments at https://github.com/nexB/vulnerablecode/blob/58d0376e7319d06387662cb393f3c39d9893088d/vulnerabilities/tests/test_import_runner.py#L201